libnavigation-core/src/main/java/com/mapbox/navigation/core/SetRoutes.kt

```diff
@@ -0,0 +1,20 @@
+package com.mapbox.navigation.core
+
+internal sealed class SetRoutes {
```

Comment:
Is there a specific reason why you replaced `SetRoutesInfo` with `SetRoutes`? It looks like it's just doing the same thing differently. But now you have more `when`s.

Reply:
They are doing the same, but they are not the same:
- `SetRoutes` is self-defined, it's not needed to hold String `Reason`
- as it's internal, now classes are data and objects
- leg index is not present by default anymore: it's not needed for CleanUp and Reroute reasons

This is deep refactoring. I wanted to remove one and define another to show up diff explicitly. 

Also, IMO (you can disagree with me 😄 ), I prefer nested sealed classes where it's possible, because they are defined under the same prefix, like 
```kotlin
sealed class Base {
    class Alpha(val value: String) : Base()
    object Beta : Base()
}

fun ololo(base: Base) {
    when(base) {
        is Base.Alpha -> TODO()
        Base.Beta -> TODO()
    }
}
```
vs 
```kotlin
sealed class Base
class AlphaBase(val value: String) : Base()
object BetaBase : Base()

fun ololo(base: Base) {
    when(base) {
        is AlphaBase -> TODO()
        BetaBase -> TODO()
    }
}
```

Reply:
1) Yes, but you still have to perform a `when` to convert it to reason. Why not just use a property?
2) SetRoutesInfo was also internal;
3) It's not present anymore by default but now you have to "invent" it for cases where it's not present.

> Also, IMO (you can disagree with me 😄 ), I prefer nested sealed classes where it's possible, because they are defined under the same prefix

I don't have a strong preference here.

Reply:
> Yes, but you still have to perform a when to convert it to reason. Why not just use a property?

Whenever we have string property or not we have to perform `when` to get the carried data. 

> SetRoutesInfo was also internal;
 
yes, so we're good to keep them aka data-object 

>It's not present anymore by default but now you have to "invent" it for cases where it's not present.

it's actually a gap in our API: we don't need LEG INDEX when it's not needed. The most common sample is CleanUp routes - we don't need a leg index here, but setting it, as API is required it. Seems not setting the leg index in the upper level is safe, to not pass it occasionally